src/libConsensus/ConsensusLeader.cpp

```diff
@@ -139,11 +139,54 @@ void ConsensusLeader::GenerateConsensusSubsets() {
     subset.commitPoints.emplace_back(m_commitPointMap.at(m_myID));
     subset.commitMap.at(m_myID) = true;
 
-    for (unsigned int j = 0; j < m_numForConsensus - 1; j++) {
-      unsigned int index = peersWhoCommitted.at(j);
-      subset.commitPointMap.at(index) = m_commitPointMap.at(index);
-      subset.commitPoints.emplace_back(m_commitPointMap.at(index));
-      subset.commitMap.at(index) = true;
+    // If DS consensus, then first subset should be of dsguard commits only.
+    // Fill in from rest if commits from dsguards < m_numForConsensus
+    if (m_DS && GUARD_MODE && (i == 0)) {
+      unsigned int subsetPeers = 1;  // myself
+      vector<unsigned int> nondsguardIndexes;
+      for (auto index : peersWhoCommitted) {
+        if (index < Guard::GetInstance().GetNumOfDSGuard()) {
+          subset.commitPointMap.at(index) = m_commitPointMap.at(index);
+          subset.commitPoints.emplace_back(m_commitPointMap.at(index));
+          subset.commitMap.at(index) = true;
+          subsetPeers++;
+          if (subsetPeers == m_numForConsensus) {
+            // got all dsguards commit
+            LOG_GENERAL(INFO, "[SubsetID: " << i << "] Got all "
+                                            << m_numForConsensus
+                                            << " commits from ds-guards");
+            break;
+          }
+        } else {
+          nondsguardIndexes.push_back(index);
+        }
+      }
+
+      // check if we fall short of commits from dsguards
+      if (subsetPeers < m_numForConsensus) {
+        // Add from rest of nondsguards commits
+        LOG_GENERAL(
+            WARNING,
+            "[SubsetID: "
+                << i << "] Did'nt got all commits from ds-guards. Expected = "
+                << m_numForConsensus << ", Received = " << subsetPeers);
+        LOG_GENERAL(WARNING, "Will consider commits from non-dsguard nodes");
```

Comment:
Maybe we can remove this message as it doesn't add much new info

Reply:
I think it adds explictly info that we are considering commits from non-dsguards. We should see this log only in WARNING situations, since inside if statement.